starboard/android/shared/video_frame_tracker.cc

```diff
@@ -14,131 +14,38 @@
 
 #include "starboard/android/shared/video_frame_tracker.h"
 
-#include <cmath>
-#include <cstdint>
-#include <cstdlib>
-#include <vector>
-
 #include "starboard/common/log.h"
-#include "starboard/common/mutex.h"
 #include "starboard/time.h"
 
 namespace starboard {
 namespace android {
 namespace shared {
-namespace {
-
-const SbTime kMaxAllowedSkew = 5000;
-
-}  // namespace
 
-SbTime VideoFrameTracker::seek_to_time() const {
-  return seek_to_time_;
-}
-
-void VideoFrameTracker::OnInputBuffer(SbTime timestamp) {
-  SB_DCHECK(thread_checker_.CalledOnValidThread());
+void VideoFrameTracker::OnInputBufferEnqueued(
+    const scoped_refptr<InputBuffer>& input_buffer) {
+  SB_DCHECK(input_buffer);
+  SB_DCHECK(media_time_provider_);
 
-  if (frames_to_be_rendered_.empty()) {
-    frames_to_be_rendered_.push_back(timestamp);
+  if (input_buffer->timestamp() < seek_to_time_) {
     return;
   }
 
-  if (frames_to_be_rendered_.size() > max_pending_frames_size_) {
-    // OnFrameRendered() is only available after API level 23.  Cap the size
-    // of |frames_to_be_rendered_| in case OnFrameRendered() is not available.
-    frames_to_be_rendered_.pop_front();
-  }
+  bool is_playing = true;
+  bool is_eos_played = true;
+  bool is_underflow = true;
+  double playback_rate = -1.0;
+  SbTime media_time = media_time_provider_->GetCurrentMediaTime(
+      &is_playing, &is_eos_played, &is_underflow, &playback_rate);
 
-  // Sort by |timestamp|, because |timestamp| won't be monotonic if there are
-  // B frames.
-  for (auto it = frames_to_be_rendered_.end();
-       it != frames_to_be_rendered_.begin();) {
-    it--;
-    if (*it < timestamp) {
-      frames_to_be_rendered_.emplace(++it, timestamp);
-      return;
-    } else if (*it == timestamp) {
-      SB_LOG(WARNING) << "feed video AU with same time stamp " << timestamp;
-      return;
-    }
+  if (input_buffer->timestamp() < media_time) {
+    dropped_frames_++;
   }
-
-  frames_to_be_rendered_.emplace_front(timestamp);
-}
-
-void VideoFrameTracker::OnFrameRendered(int64_t frame_timestamp) {
```

Comment:
Consider checking with the partners on whether the behavior of the callback are reliable on certain platforms.